src/shadowbox/server/manager_metrics.ts

```diff
@@ -15,8 +15,36 @@
 import {PrometheusClient} from '../infrastructure/prometheus_scraper';
 import {DataUsageByUser, DataUsageTimeframe} from '../model/metrics';
 
+interface Duration {
+  seconds: number;
+}
+
+interface Data {
+  bytes: number;
+}
+
+interface ServerMetricsServerEntry {
+  location: string;
+  asn: number;
+  asOrg: string;
+  tunnelTime: Duration;
+  dataTransferred: Data;
+}
+
+interface ServerMetricsAccessKeyEntry {
+  accessKeyId: number;
+  tunnelTime: Duration;
+  dataTransferred: Data;
```

Comment:
Not for this PR, but it would be helpful to add a lastAccessed here too, so we can incorporate into the UXD explorations.

It's been requested a few times, including in the recent interview and by ASL19. You can already calculate the last access time with the Prometheus metric. Just query for the bytes used by access key (or some other metric, like connections, whichever has fewer timeseries), and take the timestamp of the changes as per https://stackoverflow.com/a/54184511